_posts/2021-06-09-ejb-over-http.adoc

```diff
@@ -0,0 +1,118 @@
+[source]
+----
+/subsystem=elytron/filesystem-realm=fsRealm:add-identity(identity=quickstartUser)
+/subsystem=elytron/filesystem-realm=fsRealm:set-password(clear={password=quickstartPwd1!},identity=quickstartUser)
+/subsystem=elytron/filesystem-realm=fsRealm:add-identity-attribute(identity=quickstartUser, name=Roles, value=[guest])
+----
+
+We will also add a role decoder to be able to process the roles assigned
+
+[source]
+----
+/subsystem=elytron/simple-role-decoder=from-roles-attribute:add(attribute=Roles)
+----
+
+== Configure the Security Domain
+
+We will create a security domain that resides in the filesystem realm and will use the role decoder we created above
+
+[source]
+----
+/subsystem=elytron/security-domain=fsSD:add(realms=[{realm=fsRealm, role-decoder=from-roles-attribute}], default-realm=fsRealm,permission-mapper=default-permission-mapper)
+----
+
+We will also add the filesystem security domain mapping in the EJB3 subsystem so that the server uses the correct security domain 
+
+[source]
+----
+/subsystem=ejb3/application-security-domain=other:add(security-domain=fsSD)
+----
+
+== Configure the HTTP Transport
+
+First we must configure the http-authentication-factory for the server to authenticate over HTTP.
+Here we set the mechanism to BASIC (but could also be set to FORM), point to the fsSD security-domain, and the fsRealm filesystem-realm, and we call the newly created http-authentication-factory, example-http-auth
+
+[source]
+----
+/subsystem=elytron/http-authentication-factory=example-http-auth:add(http-server-mechanism-factory=global,security-domain=fsSD,mechanism-configurations=[{mechanism-name=BASIC,mechanism-realm-configurations=[{realm-name=fsRealm}]}])
+----
+
+Then in order to operate over HTTP, we must disable the legacy security-realm. This is done by undefining it from the http-invoker
+
+[source]
+----
+/subsystem=undertow/server=default-server/host=default-host/setting=http-invoker:undefine-attribute(name=security-realm)
+----
+
+And finally we can enable the http-invoker with the http-authentication-factory defined earlier (example-http-auth) that allows remote HTTP invocation for the EJB
+
+[source]
+----
+/subsystem=undertow/server=default-server/host=default-host/setting=http-invoker:write-attribute(name=http-authentication-factory, value=example-http-auth)
+----
+
+== Configure the Credential Store
+
+To create a credential store, we run the following command, which creates it in the path specified after the --location flag, and secures it with the password specified after the --password flag (in this case it is “StorePassword”)
+
+[source]
+----
+WILDFLY_HOME/bin/elytron-tool.sh credential-store --create --location "/PATH/TO/mycredstore.cs" --password StorePassword
+----
+
+We then create an alias to be able to access this credential store with a different username and password, in this case the username being “quickstartUser”, and the password being “quickstartPwd1!”
```

Comment:
"We then create an alias to be able to access this credential store with a different username and password"

I think this should be reworded to indicate that we're adding an alias (quickstartUser) to the credential store that will be used to hold the password that we want to use when accessing the EJB.

Reply:
Changed to
> We now create an alias that will allow us to have access to the credential store password when accessing the EJB. In this case the alias username is “quickstartUser”, and the password is “quickstartPwd1!”: